lib/experimental/interactivity-api/class-wp-interactivity-store.php

```diff
@@ -71,7 +61,9 @@ static function render() {
 		if ( empty( self::$store ) ) {
 			return;
 		}
-		$store = self::serialize();
-		echo "<script id=\"wp-interactivity-store-data\" type=\"application/json\">$store</script>";
+		echo sprintf(
```

Comment:
Have you considered using native WordPress functions like [`wp_register_script`](https://developer.wordpress.org/reference/functions/wp_register_script/) that support inline scripts? They give better control over where the code is going rendered in the document, and plugin authors can use filtering if necessary.

Reply:
[This one](https://developer.wordpress.org/reference/functions/wp_add_inline_script/)?

Reply:
Yes, that would work, you can connect it with the existing interactivity script and put it before or after depending on the actual use case.

Reply:
Perfect, thanks. Added to the [roadmap](https://github.com/WordPress/gutenberg/discussions/52904) 🙂👍️

Reply:
@gziolo, do you know if [`wp_add_inline_script`](https://developer.wordpress.org/reference/functions/wp_add_inline_script/) works with `application/json` scripts? 👀

Reply:
@DAreRodz, most likely, you would need to use WP hooks to inject that custom type.

@adamsilverstein or @westonruter, what would you advise to handle in this special case? Is it something that we should bake into `wp_add_inline_script` in WP core?

Reply:
@westonruter, thank you so much for the detailed explanation. The `id` should be enough in this case as it is standardized based on the script handle and always included for a few major WP releases.

> I'm not sure whether it makes sense to add a more direct API to change the type of such inline scripts without seeing whether there are more such use cases.

Let's see if `wp_add_inline_script` needs to be extended. If it's one of case, then the filter is more than enough.